bool is_at_desired